### PR TITLE
#2313. [Extension types] Add nullability tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — APPROVED** [review]
LGTM

[Edit: Clicked on a link that only showed part of the PR. I think we should consider the wording of one description.]

File: LanguageFeatures/Extension-types/nullability_A02_t01.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let V be an extension type of the form Name<T1, .. Ts>, and let R
+/// be the corresponding instantiated representation type. If R is non-nullable
+/// then V is a proper subtype of Object, and V is non-nullable. Otherwise, V is
+/// a proper subtype of Object?, and V is potentially nullable.
+///
+/// @description Checks that null can be assigned to an extension type at run
+/// time if its representation type is nullable
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/nullability_A02_t01.dart:11]: The description seems to be somewhat unrelated to the test?

**Review by eernstg — COMMENTED** [review]
The `??` test seems to have a description that doesn't fit well. Otherwise fine!

**Comment by sgrekhov** [comment]
Description of ?? test updated

**Review by eernstg — APPROVED** [review]
LGTM
